receiver/carbonreceiver/protocol/plaintext_parser_test.go

```diff
@@ -41,6 +41,16 @@ func Test_plaintextParser_Parse(t *testing.T) {
 				3.14,
 			),
 		},
+		{
+			line: "tst.floattimestamp 3.14 1582230020.123",
+			want: buildDoubleMetric(
+				GaugeMetricType,
+				"tst.floattimestamp",
+				nil,
+				1582230020,
+				3.14,
+			),
+		},
```

Comment:
Did you confirm that this particular test fails before the fix?

Reply:
yes, without the fix:

```
=== FAIL: protocol Test_plaintextParser_Parse/tst.floattimestamp_3.14_1582230020.123 (0.00s)
    plaintext_parser_test.go:101:
        	Error Trace:	/Users/jon.nangle/src/opentelemetry-collector-contrib/receiver/carbonreceiver/protocol/plaintext_parser_test.go:101
        	Error:      	Not equal:
        	            	expected: pmetric.Metric{orig:(*v1.Metric)(0xc0001a5fc0), state:(*internal.State)(0xc000264cdc)}
        	            	actual  : pmetric.Metric{orig:(*v1.Metric)(nil), state:(*internal.State)(nil)}

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,24 +1,4 @@
        	            	 (pmetric.Metric) {
        	            	- orig: (*v1.Metric)({
        	            	-  Name: (string) (len=18) "tst.floattimestamp",
        	            	-  Description: (string) "",
        	            	-  Unit: (string) "",
        	            	-  Data: (*v1.Metric_Gauge)({
        	            	-   Gauge: (*v1.Gauge)({
        	            	-    DataPoints: ([]*v1.NumberDataPoint) (len=1) {
        	            	-     (*v1.NumberDataPoint)({
        	            	-      Attributes: ([]v1.KeyValue) <nil>,
        	            	-      StartTimeUnixNano: (uint64) 0,
        	            	-      TimeUnixNano: (uint64) 1582230020000000000,
        	            	-      Value: (*v1.NumberDataPoint_AsDouble)({
        	            	-       AsDouble: (float64) 3.14
        	            	-      }),
        	            	-      Exemplars: ([]v1.Exemplar) <nil>,
        	            	-      Flags: (uint32) 0
        	            	-     })
        	            	-    }
        	            	-   })
        	            	-  })
        	            	- }),
        	            	- state: (*internal.State)(0)
        	            	+ orig: (*v1.Metric)(<nil>),
        	            	+ state: (*internal.State)(<nil>)
        	            	 }
        	Test:       	Test_plaintextParser_Parse/tst.floattimestamp_3.14_1582230020.123
    plaintext_parser_test.go:102:
        	Error Trace:	/Users/jon.nangle/src/opentelemetry-collector-contrib/receiver/carbonreceiver/protocol/plaintext_parser_test.go:102
        	Error:      	Not equal:
        	            	expected: false
        	            	actual  : true
        	Test:       	Test_plaintextParser_Parse/tst.floattimestamp_3.14_1582230020.123

=== FAIL: protocol Test_plaintextParser_Parse (0.00s)
```